### ISSUE TITLE
Wrong cursor position when selecting text

### ISSUE THREAD

**Issue by klayman** [geany/geany#1442]
In Geany 1.30 selecting a text with `shift`+`arrow` or `shift`+`home`/`end` leaves the last symbol under the cursor selected. For me this behavior is not apparent.

For example, pressing `shift`+`right arrow` leaves the cursor at the same position, while the symbol under the cursor gets selected. In the earlier versions (perhaps, 1.29) the same key combo selected the current symbol and moved the cursor to the next symbol.

**Comment by elextr** [comment]
Works for me.

**Comment by klayman** [comment]
It should be reproducible with themes using block cursor, e.g. "Dark", "Retro" etc.

**Comment by elextr** [comment]
oh, block caret.

Selection and display of the caret is performed by the [Scintilla](https://sourceforge.net/p/scintilla/bugs/) editing component Geany uses, please check it with Scite, the Scintilla reference editor and report upstream if not already.

Note that IIUC block caret is not supported by the Scintilla maintainer, but an external implementer, and it hasn't had much love recently.

**Comment by klayman** [comment]
Ok, I've created a ticket for Scintilla.

SciTE doesn't support block caret at all, unfortunately.

**Comment by elextr** [comment]
Looks like its deliberate, see https://sourceforge.net/p/scintilla/bugs/1924/

**Comment by klayman** [comment]
Yes, looks like so (this is the ticket I've created, by the way). There is a discussion on this topic on Google groups and I'll post my arguments for the old behavior.

**Comment by elextr** [comment]
> this is the ticket I've created, by the way
Yes thats why I put a reference to it, so people could find it easy.

Let us know what the discussion decides so we can close this or not.

**Comment by AlbertoGP** [comment]
Hi, I've been bit by this issue too when using another Scintilla-based editor, and I'm proposing a fix in that Scintilla ticket: https://sourceforge.net/p/scintilla/bugs/1924/

**Comment by codebrainz** [comment]
@AlbertoGP, FWIW you might want to make your changes to (or backport them to) Scintilla's LongTerm3 branch, since the HG tip requires bleeding edge compilers that support C++17 which was just finalized in the last couple weeks, so Scintilla-based editors you use might take a long time to get your changes if only done against HG tip.

**Comment by AlbertoGP** [comment]
After some discussion of alternatives, the simplest way for Textadept was to add a local patch that undoes the block cursor selection change that happened earlier this year:
https://foicica.com/lists/code/201710/4061.html

In case you'd like to do the same, I've made PR #1608 to first bring your `scintilla_changes.patch` up to date. The corresponding issue is #1606
Then, in another PR I would add the cursor block behaviour restoration to it.
The change that caused all this is quite simple to undo anyway: https://sourceforge.net/p/scintilla/code/ci/d29f1dad85ac3ec2a57c9341c5bd62a05c07d1e0/

**Comment by zufuliu** [comment]
https://sourceforge.net/p/scintilla/bugs/1924/
was fixed in Scintilla 4.1.8.

**Comment by elextr** [comment]
@zufuliu was it backported to the long term 3 branch?

**Comment by zufuliu** [comment]
@elextr not at the moment, but I think I will backported later.

**Comment by elextr** [comment]
Ok, Geany still uses the LTS branch, so it will get picked up after that happens.

**Comment by zufuliu** [comment]
Scintilla 3.10.0 is available now.
https://scintilla.sourceforge.io/ScintillaHistory.html
